Source/MQTTnet/Server/MqttClientSession.cs

```diff
@@ -153,6 +182,8 @@ public void Stop(MqttClientDisconnectType type)
                 }
 
                 _willMessage = null;
+
+                _workerTask?.GetAwaiter().GetResult();
```

Comment:
This will cause a deadlock as Stop is called from RunInternalAsync, and Stop if waiting for RunInternalAsync to finish ...

Stop is called in ProcessReceivedPacket, when a MqttDisconnectPacket is received